definitions/features/foreman_tasks.rb

```diff
@@ -176,6 +178,10 @@ def old_tasks_condition(state = "'stopped', 'paused'")
        "foreman_tasks_tasks.started_at < CURRENT_DATE - INTERVAL '#{MIN_AGE} days'"
   end
 
+  def paused_tasks_condition(state = "'paused'")
+    "foreman_tasks_tasks.state IN (#{state})"
```

Comment:
Here, you have taken state of tasks into consideration.
`not_paused` check calls ` feature(:foreman_tasks).paused_tasks_count(ignored_tasks)` to find specific paused tasks.  
I think, deleting only such paused tasks where result=error and not having label from  `ignored_tasks` is expected here.  Could you confirm this behavior?

Reply:
@kgaikwad  Yes it is expected that we should not delete [ignored_tasks](https://github.com/theforeman/foreman_maintain/blob/eb981e85a7b723382f5d48f054bf402065051616/definitions/checks/foreman_tasks/not_paused.rb#L26)
These tasks are meant to be running always.

Reply:
@kgaikwad @patilsuraj767 Currently we delete tasks where result=error, what about the tasks where result is pending or delayed?

Reply:
@jameerpathan111 There is another check in foreman-maintain. Check `not_running` task which is run after the check `not_paused` in upgrade workflow, it will take care of running/pending tasks.